changes/bd5679446f816eef8775bcdb34457596.yaml

```
@@ -0,0 +1,8 @@
+---
+desc: Fixed a bug where trying to modify array, data, and some comp type node properties
+  would fail or cause unexpected behavior because assigning the property to a variable
+  would not make a mutable copy of the value.
```

Maybe something like this? Currently kinda sounds like the issue occurs when you're trying to edit the node.
```suggestion
desc: Fixed a bug where assigning array, data, or some comp type node properties
  to a variable would not make a mutable copy of the value.
```